Source/SwiftLintFramework/Rules/Metrics/EnumCaseAssociatedValuesLengthRule.swift

```diff
@@ -0,0 +1,74 @@
+import SourceKittenFramework
+
+public struct EnumCaseAssociatedValuesLengthRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityLevelsConfiguration(warning: 5, error: 6)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "enum_case_associated_values_count",
+        name: "Enum Case Associated Values Count",
+        description: "Number of associated values should be low",
+        kind: .metrics,
+        nonTriggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, designation: String, contactNumber: Int)
+                case partTime(name: String, age: Int, contractEndDate: Date)
+            }
+            """
+        ],
+        triggeringExamples: [
+            """
+            enum Employee {
+                case fullTime(name: String, retirement: Date, age: Int, designation: String, contactNumber: Int)
+                case partTime(name: String, contractEndDate: Date, age: Int, designation: String, contactNumber: Int)
+            }
+            """
+        ]
+    )
+
+    public func validate(
+        file: SwiftLintFile,
+        kind: SwiftDeclarationKind,
+        dictionary: SourceKittenDictionary
+    ) -> [StyleViolation] {
+        guard kind == .enum else {
+            return []
+        }
+
+        let enumElements = substructureElements(of: dictionary, matching: .enumcase)
+            .compactMap { substructureElements(of: $0, matching: .enumelement) }
+            .flatMap { $0 }
+
+        var violations: [StyleViolation] = []
+
+        for element in enumElements {
+            guard let keyOffset = element.offset,
+                let keyName = element.name else {
+                continue
+            }
+            let enumCaseAssociatedValueCount = keyName.split(separator: ":").count - 1
+            if enumCaseAssociatedValueCount >= configuration.warning {
```

Comment:
You should validate the error threshold as well, check [FileLengthRule](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/Metrics/FileLengthRule.swift#L40) for an example.

It'd be nice to have a better message as well, like `"Enum case \(caseNameWithoutParams) should contain \(parameter.value) associated values or less: currently contains \(enumCaseAssociatedValueCount)"`